server/odc-core/src/main/java/com/oceanbase/odc/core/authority/permission/PermissionProvider.java

```diff
@@ -48,4 +48,13 @@ public interface PermissionProvider {
      */
     Permission getPermissionByResourceRoles(SecurityResource resource, Collection<String> resourceRoles);
 
+    /**
+     * @param resource {@link SecurityResource}
+     * @param actions action collection
+     * @param resourceRoles, see {@link ResourceRoleName} enums
+     * @return permission collection
+     */
+    Permission getPermissionByActionsAndResourceRoles(SecurityResource resource, Collection<String> actions,
```

Comment:
I think it is simpler to separate them. I do not see any benefits in combining them; on the contrary, it would increase the implementation cost of the this combined method.